### ISSUE TITLE
Something is broken with widget rows / Turn Lanes widget

### ISSUE THREAD

**Issue by mikehgentry** [osmandapp/OsmAnd#18755]
### Description

If you choose 'Reset to default' for the top widget panel (in e.g. bicycle or motorbike derived profiles) you get a working configuration with the Turn Lanes widget. If you then delete all the widgets from the top panel, and just add 'Turn lanes' back again, most of the time it doesn't work (you can temporarily make it work by deleting the widget and re-adding it during a navigation).

Recreating the original configuration (without choosing 'Reset to default') may be possible, but it's extremely fragile.

I've noticed that the rows are enumerated oddly - frequently I'll end up with rows 1, 3 and 5 for example. Or a default configuration will only have rows 2 and 3.

### Steps to reproduce

Reset a top widget panel to default. Delete the widgets then recreate them as best you can. I've tried in Motorcycle and Bike based profiles, they both seem to be broken.

### Actual result

It's confusing. Other sane configurations, such as just having 'Turn lanes' and nothing else, don't seem to work.

 

### Expected result

- Contiguous enumeration of rows, starting from 1.
- Possible to have Turn Lanes widget alone
- Possible to recreate base configuration without choosing 'Reset to default'.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: + 4.6.8
Android/iOS version: 11
Device model: Xcover 4s
Crash-Logs: ?
```

**Comment by mikehgentry** [comment]
Single widget:

https://github.com/osmandapp/OsmAnd/assets/1724168/f320180b-3480-4be9-a983-19cd4b48226c

Same stretch of road, only change is set top panel back to defaults:

https://github.com/osmandapp/OsmAnd/assets/1724168/deb61856-7e13-4d04-bb47-3b74430b6173

**Comment by mikehgentry** [comment]
Same conditions again, but this time I'd deleted all the widgets then manually added then back. As you can see, Lanes isn't working. During the simulated navigation I remove it and add it back, and it starts to work (though I think it breaks again eventually - I'm not completely sure about that yet).

https://github.com/osmandapp/OsmAnd/assets/1724168/c4b07992-020d-44bf-8dcd-76acb8a384cd

**Comment by yuriiurshuliak** [comment]
The bug was not reproduced. Here are the steps to reproduce:

- Choose a motorcycle profile
- Pick a multi-lane section of the highway and set the start and end points within it.
- Remove other widgets from the top panel, leaving only the Lanes widget, and review its display.
- Reset the top panel settings and simulate the same road stretch to check again.

OsmAnd~ 4.7.0#1245m, released: 2023-12-19

<table style="width:100%">
  </tr>
  <tr>
    <th>A single widget</th>
    <th>Several widgets</th>
  </tr>
  <tr>
    <td> <video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/7d6df801-1fda-4418-8c9d-c779f09dbcbb"  width="300" height="auto"/>   </td>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/3b358fbe-8ffd-48ac-ad23-6528e746147a"  width="300" height="auto"/>   </td>
  </tr>
</table>

**Comment by mikehgentry** [comment]
Hmm, puzzling.

I can't reproduce it on another phone (Planet Computers Astro Slide) here either, on either nightly or on + 4.6.8. So maybe it's got something to do with screen size or something, or maybe there's some broken configuration on my OsmAnd installation on my Xcover.

I'll try to make enough room on the Xcover to test on nightly at some point.

**Comment by mikehgentry** [comment]
This appears to be largely fixed on the XCover as of 4.6.9. I've still got one or two profiles playing up (and the non-consecutive numbers issue is still there), but most of the above discussion is probably redundant so I'll close and open another one if I figure out useful reproduction steps.

Cheers!
